### ISSUE TITLE
Allow change scale with signal

### ISSUE THREAD

**Issue by eharkins** [vega/vega#1538]
Please add the feature request label if you agree that this is one. Doesn't seem like I can add it myself.

The example I used when chatting with @domoritz [on slack](https://vega-js.slack.com/archives/C4RM5HFV1/p1548717935045800) is if I would like to change from a normal linear scale to a log scale in my plot, using a signal (in particular, one that is bound to a selection by the user). 

The way that you *can* do this now is to change the scale transform of your data depending on a signal. You can also update domains and ranges with signals. But this doesn't update the axis associated with the scale.
There are multiple levels at which this could be achieved:
* allow the `type` field of the scale to be updated with a signal
* allow the `scale` field of the axis to be updated with a signal

The former might be difficult to write durably given the specific configuration of scales depending on their `type`, as @domoritz noted in our conversation. Hopefully this is less so with the latter, but I am not sure of this. Let me know if I can be of more help to outline a way forward for this, the broader goal being configurable plots via signals, including axes, legends, etc. Thank you.

**Comment by jheer** [comment]
Thanks for the issue. I agree that this would be helpful. Binding the scale type to a signal is unlikely any time soon, as we use the scale type information to guide significant parts of the parsing and dataflow generation process. Changing the scale field of the axis (perhaps subject to some constraints) might be more feasible. I've added the enhancement label to track this going forward.

**Comment by jheer** [comment]
Yes, I'm wondering if Vega might provide a different abstraction over scales (i.e., continuous) that provides the necessary info for later dataflow generation, while enabling transformation (log, sqrt, etc) as a free variable that can be controlled by a signal. However, this could result in breaking changes so we'll have to do some design work first and then decide.

**Comment by roying** [comment]
@eharkins   Instead of changing scale type, consider using Vega's power scale ("pow") with exponent that changes with bound signal input. See example at:
      https://groups.google.com/forum/#!topic/vega-js/1JdIFH6o9z4
